ffmpeg_normalize/_streams.py

```diff
@@ -320,123 +321,94 @@ def parse_loudnorm_stats(self) -> Iterator[float]:
-                    continue
-                if key == "normalization_type":
-                    loudnorm_stats[key] = loudnorm_stats[key].lower()
-                # handle infinite values
-                elif float(loudnorm_stats[key]) == -float("inf"):
-                    loudnorm_stats[key] = -99
-                elif float(loudnorm_stats[key]) == float("inf"):
-                    loudnorm_stats[key] = 0
-                else:
-                    # convert to floats
-                    loudnorm_stats[key] = float(loudnorm_stats[key])
-
-            return cast(EbuLoudnessStatistics, loudnorm_stats)
-        except Exception as e:
-            raise FFmpegNormalizeError(
-                f"Could not parse loudnorm stats; wrong JSON format in string: {e}"
-            )
+            if stream_index < 0:
+                if m := _loudnorm_pattern.match(line):
+                    loudnorm_start = index + 1
+                    stream_index = int(m.group(1))
+            else:
+                if line.startswith("}"):
+                    loudnorm_end = index + 1
+                    loudnorm_data = "\n".join(output_lines[loudnorm_start:loudnorm_end])
+
+                    try:
+                        loudnorm_stats = json.loads(loudnorm_data)
+
+                        _logger.debug(
+                            f"Loudnorm stats for stream {stream_index} parsed: {loudnorm_data}"
+                        )
+
+                        for key in [
+                            "input_i",
+                            "input_tp",
+                            "input_lra",
+                            "input_thresh",
+                            "output_i",
+                            "output_tp",
+                            "output_lra",
+                            "output_thresh",
+                            "target_offset",
+                            "normalization_type",
+                        ]:
+                            if key not in loudnorm_stats:
+                                continue
+                            if key == "normalization_type":
+                                loudnorm_stats[key] = loudnorm_stats[key].lower()
+                            # handle infinite values
+                            elif float(loudnorm_stats[key]) == -float("inf"):
+                                loudnorm_stats[key] = -99
+                            elif float(loudnorm_stats[key]) == float("inf"):
+                                loudnorm_stats[key] = 0
+                            else:
+                                # convert to floats
+                                loudnorm_stats[key] = float(loudnorm_stats[key])
+
+                        result[stream_index] = cast(EbuLoudnessStatistics, loudnorm_stats)
+                    except Exception as e:
+                        raise FFmpegNormalizeError(
+                            f"Could not parse loudnorm stats; wrong JSON format in string: {e}"
```

Comment:
This is where it fails in 

```
python3 -m ffmpeg_normalize test/test.mp4 -pr -f
```

because it apparently tries to parse both loudnorm JSON outputs, so the `lounorm_end` detection seems to not work correctly.